File: klinara-ios/klinara-ios/DesignSystem/KlinaraMetrics.swift
import SwiftUI

/// Ölçü ve hareket sabitleri. 4pt grid; cömert boşluk tasarımın
/// "clinical breathing room" ilkesinin doğrudan karşılığıdır.
enum KlinaraMetrics {

    // MARK: Boşluk (4pt grid)

    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 16
    static let lg: CGFloat = 24
    static let xl: CGFloat = 32
    static let xxl: CGFloat = 40

    /// Ekran yatay kenar boşluğu — her auth ekranında aynı.
    static let screenInset: CGFloat = 24
    /// Başlık bloğu ile ilk alan arası.
    static let headerToContent: CGFloat = 32
    /// Bağımsız bölümler arası.
    static let sectionGap: CGFloat = 40

    // MARK: Yarıçap

    static let controlRadius: CGFloat = 12
    static let cardRadius: CGFloat = 16

    // MARK: Kontrol ölçüleri

    static let controlHeight: CGFloat = 52
    static let fieldHeight: CGFloat = 52
    static let borderWidth: CGFloat = 1
    static let focusBorderWidth: CGFloat = 1.5

    // MARK: Hareket
    //
    // Sakin ve ölçülü. Uyarıcı, zıplayan ya da dikkat çeken animasyon yok —
    // marka kişiliği "calm, authoritative".

    /// Akış adımları arası geçiş.
    static let stepTransition: Animation = .smooth(duration: 0.28)
    /// Hata görünüp kaybolurken.
    static let feedback: Animation = .snappy(duration: 0.2)
}

File: klinara-android/app/src/main/java/com/klinara/android/designsystem/KlinaraMetrics.kt
package com.klinara.android.designsystem

import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.FiniteAnimationSpec
import androidx.compose.animation.core.tween
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Shapes
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Ölçü ve hareket sabitleri. 4dp grid; cömert boşluk tasarımın
 * "clinical breathing room" ilkesinin doğrudan karşılığıdır.
 *
 * CompositionLocal DEĞİL, düz `object`: temaya göre değişmiyorlar. CompositionLocal
 * saf dolaylılık olur ve `Modifier.padding(KlinaraMetrics.md)`'yi composable olmayan
 * yardımcılarda yazmayı engellerdi.
 */
object KlinaraMetrics {
    // --- Boşluk (4dp grid) ---
    val xs: Dp = 4.dp
    val sm: Dp = 8.dp
    val md: Dp = 16.dp
    val lg: Dp = 24.dp
    val xl: Dp = 32.dp
    val xxl: Dp = 40.dp

    /** Ekran yatay kenar boşluğu — her auth ekranında aynı. */
    val screenInset: Dp = 24.dp

    /** Başlık bloğu ile ilk alan arası. */
    val headerToContent: Dp = 32.dp

    /** Bağımsız bölümler arası. */
    val sectionGap: Dp = 40.dp

    // --- Yarıçap ---
    val controlRadius: Dp = 12.dp
    val cardRadius: Dp = 16.dp

    // --- Kontrol ölçüleri ---
    val controlHeight: Dp = 52.dp
    val fieldHeight: Dp = 52.dp
    val borderWidth: Dp = 1.dp
    val focusBorderWidth: Dp = 1.5.dp

    /**
     * Android erişilebilirlik tabanı. iOS'ta karşılığı 44pt; Material 48dp istiyor ve
     * §7.3 bunu batch kapanış ölçütü yapıyor.
     */
    val minTouchTarget: Dp = 48.dp

    // --- Hareket ---
    // Sakin ve ölçülü. Zıplayan, dikkat çeken animasyon YOK — marka kişiliği
    // "calm, authoritative". Süreler iOS ile birebir.

    /** Akış adımları arası geçiş (iOS `.smooth(duration: 0.28)`). */
    val stepTransition: FiniteAnimationSpec<Float> =
        tween(durationMillis = 280, easing = CubicBezierEasing(0.4f, 0f, 0.2f, 1f))

    /** Hata görünüp kaybolurken (iOS `.snappy(duration: 0.2)`). */
    val feedback: FiniteAnimationSpec<Float> =
        tween(durationMillis = 200, easing = CubicBezierEasing(0.3f, 0f, 0.1f, 1f))

    const val STEP_TRANSITION_MILLIS: Int = 280
    const val FEEDBACK_MILLIS: Int = 200

    /** Basılı hâl: tüm kontrole uygulanan alfa, renk değil. */
    const val PRESSED_ALPHA: Float = 0.72f
}

/** Material3'ün taşıyıcı olarak kullandığı şekiller — bizim yarıçaplarımızla. */
internal val KlinaraShapes =
    Shapes(
        extraSmall = RoundedCornerShape(KlinaraMetrics.sm),
        small = RoundedCornerShape(KlinaraMetrics.controlRadius),
        medium = RoundedCornerShape(KlinaraMetrics.cardRadius),
        large = RoundedCornerShape(KlinaraMetrics.cardRadius),
        extraLarge = RoundedCornerShape(KlinaraMetrics.xl),
    )
